src/main/kotlin/com/theagilemonkeys/kava/utils/kotlin/async/generator/CompletableFutureReplacementGenerator.kt

```diff
@@ -0,0 +1,44 @@
+package com.theagilemonkeys.kava.utils.kotlin.async.generator
+
+import com.squareup.kotlinpoet.FileSpec
+import com.squareup.kotlinpoet.FunSpec
+import com.squareup.kotlinpoet.ParameterSpec
+import com.squareup.kotlinpoet.ParameterizedTypeName.Companion.parameterizedBy
+import com.squareup.kotlinpoet.TypeName
+import com.squareup.kotlinpoet.asTypeName
+import kotlinx.coroutines.GlobalScope
+import java.util.concurrent.CompletableFuture
+
+class CompletableFutureReplacementGenerator : ReplacementGenerator() {
+    override fun initialise(fileSpec: FileSpec.Builder) {
+        fileSpec.addImport("kotlinx.coroutines.future", "future")
+    }
+
+    override fun suspendFunction(
+        name: String,
+        parameters: List<ParameterSpec>,
+        returnTypeName: TypeName
+    ): FunSpec = FunSpec.builder(name)
+        .addParameters(parameters)
+        .returns(futureType.parameterizedBy(Unit::class.asTypeName()))
+        .beginControlFlow("return %T.future {", GlobalScope::class)
```

Comment:
Is GlobalScope a good option? I've read that it can cause memory leaks here -> https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/-global-scope/

Reply:
That's a fair point, but we need to provide some scope. Perhaps this would be something we could drive by the annotation, let people provide a scope.